src/sync/remote_mongo_collection.cpp

```diff
@@ -18,3 +18,505 @@
+        }
+        
+        auto args = nlohmann::json( {{"arguments", nlohmann::json::array({base_args} ) }} );
+                
+        m_service->call_function("find",
+                                 args.dump(),
+                                 [completion_block](util::Optional<AppError> error, util::Optional<std::string> value) {
+            handle_response(error, value, completion_block);
+        });
+    } catch (const std::exception& e) {
+        return completion_block({}, AppError(make_error_code(JSONErrorCode::malformed_json), e.what()));
+    }
+}
+
+void RemoteMongoCollection::find(const std::string& filter_json,
+                                 std::function<void(std::string, util::Optional<AppError>)> completion_block)
+{
+    find(filter_json, {}, completion_block);
+}
+
+void RemoteMongoCollection::find_one(const std::string& filter_json,
+                                 RemoteFindOptions options,
+                                 std::function<void(std::string, util::Optional<AppError>)> completion_block)
+{
+        
+    try {
+        auto base_args = m_base_operation_args;
+        base_args.push_back({ "query", nlohmann::json::parse(filter_json) });
+        
+        if (options.limit) {
+            base_args.push_back({ "limit", options.limit.value() });
+        }
+        
+        if (options.projection_json) {
+            base_args.push_back({ "project", nlohmann::json::parse(options.projection_json.value()) });
+        }
+        
+        if (options.projection_json) {
+            base_args.push_back({ "sort", nlohmann::json::parse(options.sort_json.value()) });
+        }
+        
+        auto args = nlohmann::json( {{"arguments", nlohmann::json::array({base_args} ) }} );
+
+        m_service->call_function("findOne",
+                                 args.dump(),
+                                 [completion_block](util::Optional<AppError> error, util::Optional<std::string> value) {
+            handle_response(error, value, completion_block);
+        });
+    } catch (const std::exception& e) {
+        return completion_block({}, AppError(make_error_code(JSONErrorCode::malformed_json), e.what()));
+    }
+}
+
+void RemoteMongoCollection::find_one(const std::string& filter_json,
+                                 std::function<void(std::string, util::Optional<AppError>)> completion_block)
+{
+    find_one(filter_json, {}, completion_block);
+}
+
+void RemoteMongoCollection::insert_one(const std::string& value_json,
+                                       std::function<void(std::string, util::Optional<AppError>)> completion_block)
+{
+    try {
+        auto base_args = m_base_operation_args;
+        auto document_json = nlohmann::json::parse(value_json);
+        base_args.push_back({ "document", document_json });
+        auto args = nlohmann::json( {{"arguments", nlohmann::json::array({base_args} ) }} );
```

Comment:
The whitespace whenever braced init is involved is inconsistent both within this PR and with the rest of the codebase. The style we use elsewhere is to treat them like parentheses used for function calls, so no spaces before or after the brace (so this particular line would be `nlohmann::json({{"arguments", nlohmann::json::array({base_args})}});`)

Reply:
removed whitespace